src/main/scala/deepSum.sc

```diff
@@ -0,0 +1,22 @@
+def sum(list: List[Int]): Int = list match {
+  case Nil => 0
+  case head :: tail =>
+
+    head + sum(tail)
+}
+
+println(sum(List(1,2,3,4)))
+
+
+def deepsum(list: List[List[Int]]): Int = list match {
+  case Nil => 0
+  case head :: tail =>
+
+    sum(head) + deepsum(tail)
```

Comment:
Very smart, could you do another one without using `sum` or other helper methods/functions?

Reply:
I did some experiments but was not able to achieve the goal (sum of iterative loop). I did some changes and added some line of code. I am stuck now because, First .. I was not able to merge the changes that you proposed  *https://github.com/Jitenraiqa/scala-jiten/pull/2*  :( . Second.. I pushed the changes but not able to see them. I have them in my local but not sure what I am doing wrong. 

I will arrange an appointment to discuss about these small issues.

Reply:
I added this line of code. Idea was not to use any pre built functions 

***
def deepSum(xs: List[List[Int]]): Int = xs match {
  case List() => 0
  case (y :: yss) => y.head + deepSum(yss.tail)
  case ((y :: yss) :: xss) =>  (y :: yss).head + deepSum(xss.tail)
  case ((y :: yss) :: (x :: xss)) =>  (y :: yss).head + deepSum(x :: xss.tail)
  case _ => "out of scope"
***

Also, I was thinking to use **flatten** but again it was complaining about int --> string conversions. I will add these points in the discussions.